2019-08-03-q-and-as-about-nix-for-haskellers.md

```diff
@@ -0,0 +1,692 @@
+title = Questions and Answers about Nix for Haskellers
+tags = haskell, nixos
+summary = Overview of Nix for Haskell development
+draft = true
+
+------------------------------------------------------
+
+Last week Dmitrii Kovanikov ([@ChShersh](https://twitter.com/ChShersh)) posted
+a [request](https://twitter.com/ChShersh/status/1154782785967542272?s=20) on
+Twitter asking if anyone had the time to answer some questions about how to
+understand Nix as it applies to Haskell programming.
+
+Dmitrii is working on a scaffolding tool for Haskell projects called
+[summoner](https://github.com/kowainik/summoner).  There is an
+[open PR](https://github.com/kowainik/summoner/pull/283) adding Nix support.
+However, Dmitrii doesn't have any Nix experience, so he'd like to learn more
+about using Nix for Haskell development before tackling the PR.
+
+I've been using Nix for about a year and half, both at work and on my own
+projects.  I think there is a big lack of documentation surrounding Nix and
+Haskell, so I told him I'd be able to answer any questions he had.
+
+He wrote me back with a _great_ list of questions.  I thought it would be an
+injustice to the Haskell community if I didn't share my answers, so I decided
+to turn his questions and my answers into an article.
+
+This article is mainly for Haskellers that have heard about Nix (on Twitter,
+Reddit, YouTube, etc), know about its advantages, but haven't actually played
+around with it.  It is mainly a **high-level overview** of how Nix works with
+Haskell.  It is not a hands-on tutorial or walk-through.
```

Comment:
About that: what do you think about writing minimal possible `.nix` configuration file for some basic package and explain it line-by-line? And then you can expand blog post from that point to tell, what else can be improved.

For example, what I learned so far, is that you can use `cabal2nix` tool to generate `default.nix` file from the `.cabal` file. But probably the generated file contains much more than you need?.. And probably it's possible to write simpler version by hands?.. Also, there'is a problem with that approach is that you need to regenerate the file each time you change critical metadata in the `.cabal` file. 

So you see what I mean. You start with something simple which makes sense on its own. And then you add more layers and expand, building intuition on what is already learned. I don't propose to vendor full Haskell + Nix tutorial in this blog post. But maybe having very small example would be nice.